dns-integrator-operator/src/charm.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""DNS Integrator Charm."""
+
+import logging
+import typing
+import uuid
+
+import ops
+from charms.bind.v0 import dns_record
+
+logger = logging.getLogger(__name__)
+
+# the following UUID is used as namespace for the uuidv5 generation
+UUID_NAMESPACE = uuid.UUID("c53fe32f-6486-4f24-9417-b79b2cb596c4")
+
+
+class DnsIntegratorCharm(ops.CharmBase):
+    """Charm the service."""
+
+    def __init__(self, *args: typing.Any):
+        """Construct.
+
+        Args:
+            args: Arguments passed to the CharmBase parent constructor.
+        """
+        super().__init__(*args)
+        self.dns_record = dns_record.DNSRecordRequires(self)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.start, self._on_start)
+        self.framework.observe(self.on.collect_unit_status, self._on_collect_status)
+
+    def _on_start(self, _: ops.StartEvent) -> None:
+        """Handle start.
+
+        This hook is here to display some status after installation via collect_status().
+        """
+
+    def _on_collect_status(self, _: ops.CollectStatusEvent) -> None:
+        """Handle collect status event."""
+        if str(self.config["requests"]) == "":
+            self.unit.status = ops.BlockedStatus("Waiting for some configuration")
+            return
+
+        has_integration = False
+        for __ in self.model.relations[self.dns_record.relation_name]:
+            has_integration = True
+            break
+        if not has_integration:
+            self.unit.status = ops.BlockedStatus("Waiting for integration")
+            return
+
+        self.unit.status = ops.ActiveStatus()
+
+    def _on_config_changed(self, _: ops.ConfigChangedEvent) -> None:
+        """Handle changes in configuration by updating the relation databag."""
+        self.unit.status = ops.MaintenanceStatus("Configuring charm")
+        self._update_relations()
+        self.unit.status = ops.ActiveStatus()
```

Comment:
If you're managing status through collect status you should let the hook method manage the final status

Reply:
Fixed